app/src/sharedTest/java/org/oppia/android/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -131,399 +115,170 @@ class ProfileChooserFragmentTest {
-      intended(hasComponent(PinPasswordActivity::class.java.name))
-      onView(withId(R.id.input_pin)).perform(typeText("12345"))
-      intended(hasComponent(HomeActivity::class.java.name))
-      onView(isRoot()).perform(pressBack())
-      onView(withText(R.string.home_activity_back_dialog_exit)).perform(click())
-      intended(hasComponent(ProfileChooserActivity::class.java.name))
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          0, R.id.profile_last_visited
-        )
-      ).check(
-        matches(
-          isDisplayed()
-        )
-      )
+      it.onActivity {
+        data.observeForever { }
+      }
+      testCoroutineDispatchers.runCurrent()
       onView(
         atPositionOnView(
           R.id.profile_recycler_view,
-          0, R.id.profile_last_visited
-        )
-      ).check(
-        matches(
-          withText(
-            String.format(
-              getResources().getString(R.string.profile_last_used) + " just now"
-            )
-          )
+          0,
+          R.id.profile_last_visited
         )
+      ).check(matches(isDisplayed()))
+      matchStringOnProfileListItem(
+        position = 0,
+        targetView = R.id.profile_last_visited,
+        stringToMatch = "${context.getString(R.string.profile_last_used)} just now"
       )
     }
   }
 
   @Test
-  // TODO(#973): Fix ProfileChooserFragmentTest
-  @Ignore
-  fun testProfileChooserFragment_initializeProfiles_changeConfiguration_checkProfilesLastVisitedTimeIsShown() { // ktlint-disable max-length-line
-    profileTestHelper.initializeProfiles()
+  fun testProfileChooserFragment_afterVisitingHomeActivity_changeConfiguration_showsJustNowText() {
+    val data = profileTestHelper.initializeProfiles()
     launch<ProfileChooserActivity>(createProfileChooserActivityIntent()).use {
-      onView(
-        atPosition(
-          R.id.profile_recycler_view,
-          0
-        )
-      ).perform(click())
-      intended(hasComponent(PinPasswordActivity::class.java.name))
-      onView(withId(R.id.input_pin)).perform(typeText("12345"))
-      intended(hasComponent(HomeActivity::class.java.name))
-      onView(isRoot()).perform(pressBack())
-      onView(withText(R.string.home_activity_back_dialog_exit)).perform(click())
-      intended(hasComponent(ProfileChooserActivity::class.java.name))
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          0, R.id.profile_last_visited
-        )
-      ).check(
-        matches(
-          isDisplayed()
-        )
-      )
-      onView(
-        atPositionOnView(
-          R.id.profile_recycler_view,
-          0, R.id.profile_last_visited
-        )
-      ).check(
-        matches(
-          withText(
-            String.format(
-              getResources().getString(R.string.profile_last_used) + " just now"
-            )
-          )
-        )
-      )
+      it.onActivity {
+        data.observeForever { }
+      }
```

Comment:
This is a bit obscure. Suggest putting it behind a helper method, e.g. ``waitForOperationToComplete()`` that takes a live data, calls runCurrent(), then removes the observer. Then, update all methods that need this to call the new helper.

Please make sure the helper clearly documents _why_ the observeForever is needed & why it works. Also, please update the documentation for all the helpers in ProfileTestHelper that could result in this issue with instructions on how to make sure the operations finish (e.g. by adding an observer & calling runCurrent()).

Reply:
Done

Reply:
Resolving on behalf of Ben